### DISCUSSION TITLE
Error after updating env: "missing symbol called"

### DISCUSSION THREAD

**Discussion by marinasessim** [idaholab/moose#21703] [Q&A General] [answered]
Hello, 

I'm getting an error on an older app after updating the moose environment. I get this error when I run any input file or the tests.

"dyld[xxxxx]: missing symbol called"

I cleaned up before recompiling by running "make clobberall", which was marked as the solution for this error on a similar thread. 

I'm attaching the outputs of what I think is the equivalent of "ldd executable" on a mac (otool -L). 

Please let me know if there is any other information that can help identify the source. The application is a few months old, please let me know if there are deprecated functions/syntax/anything that could be causing this error. 

Thanks for your attention!

[macaw-opt_output.txt](https://github.com/idaholab/moose/files/9200821/macaw-opt_output.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hello

`make clobberall` is often insufficient
we tend to use `git clean -xfd` these days
please make sure to commit ALL your work before running this command. 

What is macaw? What does it do?
We have a controlled app with the same name, so I'd just like to check

Guillaume

**Reply by marinasessim** [reply]
Hi Guillaume, 

Unfortunately "git clean -xfd" did not do the trick either. 

I updated the environment "conda update --all" and rebased moose again. It compiles and no test fails. I did notice a warning this time, it shows up at compilation time (same for Macaw):

(moose) [msessim][~/projects/macaw]> make -j 4
Checking if conda packages exist...
Checking if libmesh version is up to date...
/Users/msessim/projects/moose/framework/moose.mk:34: The moose-libmesh conda package build number does match the current version of MOOSE. Please run "conda update --all" in your MOOSE environment.
Checking if petsc version is up to date...
/Users/msessim/projects/moose/framework/moose.mk:50: The moose-petsc conda package build number does match the current version of MOOSE. Please run "conda update --all" in your MOOSE environment.
(...)

I had just performed "conda update --all". Is there any other package update that I am missing?

I still get the "missing symbol called" when I run the tests in Macaw, or any input file. Let me know if there is any info that will help here.

Thanks for your interest in Macaw. I created Macaw (https://github.com/marinasessim/macaw) to share some of the work I did in grad school with Dr.Tonks (https://doi.org/10.1016/j.commatsci.2021.111156). We used phase-field to simulate the oxidation of carbon fibers in a material that is used in heat shields of spacecraft. I did not know we had another Macaw application, bummer! 

Marina

**Reply by GiudGiud** [reply]
Hello

So conda update --all can fail to update sometimes. What versions do you have of the packages?
```
moose-libmesh             2022.06.06              build_3    https://conda.software.inl.gov/public
moose-mpich               4.0.2                   build_0    https://conda.software.inl.gov/public
moose-petsc               3.16.5                  build_6    https://conda.software.inl.gov/public
moose-tools               2022.06.16       py39h5a31bc0_0    https://conda.software.inl.gov/public
```
if you have something older than those, you may be hitting this. 
In this case, it's easiest to first try `mamba update moose-libmesh moose-petsc` then if this fails, wipe the conda environment and reinstall them

Guillaume

**Reply by marinasessim** [reply]
Hi Guillaume, sorry for the delay. This was the version of the aforementioned packages:

moose-libmesh             2022.06.06              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0            py39hc1bf809_7    https://conda.software.inl.gov/public
moose-mpich               4.0.1                   build_2    https://conda.software.inl.gov/public
moose-petsc               3.16.5                  build_3    https://conda.software.inl.gov/public
moose-tools               2022.04.18       py39hf94b855_0    https://conda.software.inl.gov/public

I'm in the process of updating the moose env now (starting over since neither "conda update --all" or the "mamba update moose-libmesh moose-petsc" worked). I asked a colleague (@timcalvert) to try it out, he has a fresh environment with the updated packages and the "missing symbol called" problem persists. We also tried in our linux cluster in UF and same is true there. Do you have any other suggestions about what could be happening? Let me know if there is anything I can share to help shed some light. 

Thanks for your help! 
Marina

**Reply by GiudGiud** [reply]
the "missing symbol called" error suggests there are old compiled objects in the repository. I would use the `git clean -xfd` routine in the moose/ folder and try again. Note that you need to have saved / committed all your work before using that.

The package versions are slightly outdated. You probably got newer versions when re-creating the environment from scratch?

Guillaume

**Reply by marinasessim** [reply]
Hello, 

I removed the old environment and created a new one. MOOSE tests were compiled and ran with no problems. The packages are now updated:

```
moose-libmesh             2022.08.05              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.1.0               h30e85a9_11    https://conda.software.inl.gov/public
moose-mpich               4.0.2                   build_1    https://conda.software.inl.gov/public
moose-petsc               3.16.6                  build_0    https://conda.software.inl.gov/public
moose-tools               2022.07.18      py310hcffbc79_0    https://conda.software.inl.gov/public
```

I performed both `git clean -xfd` and `make clobberall` in the macaw dir. Macaw was recompiled, but the tests still hit the "missing symbol called" error. I'm attaching the outputs of `./run_tests`. I will try to run it in debugging mode next. 

[tests_missing_symbol_out.txt](https://github.com/idaholab/moose/files/9338882/tests_missing_symbol_out.txt)

**Reply by GiudGiud** [reply]
Is MOOSE a submodule of macaw in your build?

if so you may want to use 
git submodule for each git clean -xfd in the macaw dir as well

**Reply by marinasessim** [reply]
Hi Guillaume,

I don't think I have moose as a submodule, there is no /moose in the macaw dir and I set `export MOOSE_DIR="$HOME/projects/moose"` in the bash_profile. 

I just tried ` git submodule foreach git clean -xfd` in addition to `git clean -xfd` and `make clobberall` in the macaw dir.  Same error while running tests ("missing symbol called").

**Reply by GiudGiud** [reply]
if it s not a submodule, can you clean moose with the `git clean -xfd` command in projects/moose?

missing symbol called could also be a missing function definition I think
